lib/utils.js

```diff
@@ -631,6 +631,9 @@ exports.getValue = function(path, obj, map) {
 const mapGetterOptions = Object.freeze({ getters: false });
 
 function getValueLookup(obj, part) {
+  if (part === '$*' && obj instanceof Map) {
+    return obj;
+  }
```

Comment:
Maybe this should get a test?

Reply:
This code is covered by the tests added in the PR, albeit indirectly.